File: authentication-service/models/user_model.go
package models

import "time"

type UserModel struct {
    Email     string    `bson:"email"`
    Name      string    `bson:"name"`
    Password  string    `bson:"password"`
    CreatedAt time.Time `bson:"createdDate"`
    Type      string    `bson:"accountType"`
}

File: backend/user-backend/src/model/user-model.js
import mongoose from "mongoose";

export const DEFAULT_IMAGE = 'DEFAULT';

const Schema = mongoose.Schema;
const UserModelSchema = new Schema({
  username: {
    type: String,
    required: true,
    unique: true,
  },
  email: {
    type: String,
    required: true,
    unique: true,
  },
  password: {
    type: String,
    required: true,
  },
  profileImage: {
    type: String,
    required: true,
    default: DEFAULT_IMAGE,
  },
  isAdmin: {
    type: Boolean,
    required: true,
    default: false,
  },
  isVerified: {
    type: Boolean,
    required: true,
    default: false,
  },
  createdAt: {
    type: Date,
    default: Date.now, // Setting default to the current date/time
  },
});

export default mongoose.model("users", UserModelSchema);
